kmath-tensors/src/commonMain/kotlin/space/kscience/kmath/tensors/core/DoubleTensorAlgebra.kt

```diff
@@ -706,14 +705,377 @@ public open class DoubleTensorAlgebra :
+        var p_old = zeros(ShapeND(intArrayOf(Npar, 1))).as2D()    // previous set of parameters
+        var y_old = zeros(ShapeND(intArrayOf(Npnt, 1))).as2D()    // previous model, y_old = y_hat(t;p_old)
+        var X2 = 1e-3 / eps                                       // a really big initial Chi-sq value
+        var X2_old = 1e-3 / eps                                   // a really big initial Chi-sq value
+        var J = zeros(ShapeND(intArrayOf(Npnt, Npar))).as2D()     // Jacobian matrix
+        val DoF = Npnt - Npar                                     // statistical degrees of freedom
+
+        var corr_p = 0
+        var sigma_p = 0
+        var sigma_y = 0
+        var R_sq = 0
+        var cvg_hist = 0
+
+        if (length(t) != length(y_dat)) {
+            // println("lm.m error: the length of t must equal the length of y_dat")
+            val length_t = length(t)
+            val length_y_dat = length(y_dat)
+            X2 = 0.0
+
+            corr_p = 0
+            sigma_p = 0
+            sigma_y = 0
+            R_sq = 0
+            cvg_hist = 0
+        }
+
+        var weight = weight_input
+        if (nargin <  5) {
+            weight = fromArray(ShapeND(intArrayOf(1, 1)), doubleArrayOf((y_dat.transpose().dot(y_dat)).as1D()[0])).as2D()
+        }
+
+        var dp = dp_input
+        if (nargin < 6) {
+            dp = fromArray(ShapeND(intArrayOf(1, 1)), doubleArrayOf(0.001)).as2D()
+        }
+
+        var p_min = p_min_input
+        if (nargin < 7) {
+            p_min = p
+            p_min.abs()
+            p_min = p_min.div(-100.0).as2D()
+        }
+
+        var p_max = p_max_input
+        if (nargin < 8) {
+            p_max = p
+            p_max.abs()
+            p_max = p_max.div(100.0).as2D()
+        }
+
+        var c = c_input
+        if (nargin < 9) {
+            c = fromArray(ShapeND(intArrayOf(1, 1)), doubleArrayOf(1.0)).as2D()
+        }
+
+        var opts = opts_input
+        if (nargin < 10) {
+            opts = doubleArrayOf(3.0, 10.0 * Npar, 1e-3, 1e-3, 1e-1, 1e-1, 1e-2, 11.0, 9.0, 1.0)
+        }
+
+        val prnt          = opts[0]                // >1 intermediate results; >2 plots
+        val MaxIter       = opts[1].toInt()        // maximum number of iterations
+        val epsilon_1     = opts[2]                // convergence tolerance for gradient
+        val epsilon_2     = opts[3]                // convergence tolerance for parameters
+        val epsilon_3     = opts[4]                // convergence tolerance for Chi-square
+        val epsilon_4     = opts[5]                // determines acceptance of a L-M step
+        val lambda_0      = opts[6]                // initial value of damping paramter, lambda
+        val lambda_UP_fac = opts[7]                // factor for increasing lambda
+        val lambda_DN_fac = opts[8]                // factor for decreasing lambda
+        val Update_Type   = opts[9].toInt()        // 1: Levenberg-Marquardt lambda update
```

Comment:
Unconventional formatting.

Reply:
Underscores in the local variables' names are also disregarded.

Reply:
fixed